tests/chainerx_tests/unit_tests/routines_tests/test_evaluation.py

```diff
@@ -0,0 +1,67 @@
+import chainer
+from chainer import functions as F
+import numpy
+
+import chainerx
+
+from chainerx_tests import op_utils
+
+
+_in_out_eval_dtypes = [
+    (('float16', 'int16'), 'float32'),
+    (('float32', 'int32'), 'float32'),
+    (('float64', 'int64'), 'float64'),
+    (('float32', 'int16'), 'float32'),
+    (('float64', 'int16'), 'float64'),
+    (('float64', 'int32'), 'float64'),
+]
+
+
+class EvalBase(op_utils.ChainerOpTest):
+
+    def generate_inputs(self):
+        x_dtype, t_dtype = self.in_dtypes
+        y = numpy.random.uniform(-1, 1, self.x_shape).astype(x_dtype)
+        targ = numpy.random.randint(
+            3, size=self.t_shape).astype(t_dtype)
+        return y, targ
+
+    def forward_chainerx(self, inputs):
+        return self.forward_xp(inputs, chainerx)
+
+    def forward_chainer(self, inputs):
+        return self.forward_xp(inputs, F)
+
+    def forward_xp(self, inputs, xp):
+        raise NotImplementedError(
+            'Op test implementation must override `forward_xp`.')
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product([
+        chainer.testing.from_pytest_parameterize(
+            'x_shape,t_shape', [
+                ((10, 3), (10,)),
+                ((10, 3, 1), (10,)),
+                ((10, 3, 1, 1), (10,)),
+                ((10, 3, 5), (10, 5)),
+                ((10, 3, 5, 4), (10, 5, 4)),
+                ((10, 3, 5, 4, 1), (10, 5, 4)),
+                ((10, 3, 5, 4, 1, 1), (10, 5, 4))
+            ]),
+        chainer.testing.from_pytest_parameterize(
+            'in_dtypes,out_dtype', _in_out_eval_dtypes),
```

Comment:
Is the `out_dtype` read?

Reply:
It isn't being read. Why is this so?

Reply:
With `ChainerOpTest`, you're checking the dtype against Chainer's output dtypes, which by the way differ from the unused expected outputs that you've defined above. How did you choose the expected output dtypes in `_in_out_eval_dtypes`?

Reply:
To answer your question, it's not being read because that's how `ChainerOpTest` is defined and implemented.

Reply:
Made the fixes. Thanks for your reply.